src/mcts/node.h

```diff
@@ -152,15 +243,46 @@ class Node {
   // Returns n = n_if_flight.
   int GetNStarted() const { return n_ + n_in_flight_; }
   // Returns node eval, i.e. average subtree V for non-terminal node and -1/0/1
-  // for terminal nodes.
-  float GetQ() const { return q_; }
+  // for terminal nodes (read from edge).
+  float GetQ() const;
 
   // Returns whether the node is known to be draw/lose/win.
-  bool IsTerminal() const { return is_terminal_; }
+  bool IsTerminal() const {
```

Comment:
With need to access those from Node, it surely feels like certainty is a property of a node, not edge.
(also it's not efficient as `GetOwnEdge()` is evaluated multiple times, but even without that it doesn't look like it's the right way to do it).

Reply:
This pertains to the edge vs. node centricity of the code. See comment above.